src/Avalonia.Controls/TextBoxTextInputMethodClient.cs

```diff
@@ -177,7 +177,7 @@ public void SetPreeditText(string? preeditText)
 
             var text = GetText(preeditText);
 
-            _presenter._text = text;
+            _presenter.SetCurrentValue(TextPresenter.TextProperty, text);
 
             _presenter.PreeditText = preeditText;
```

Comment:
Why aren't we using SetCurrentValue here?

Reply:
Do you mean "why ARE we using..."?

Reply:
I think it's about `PreeditText = preeditText`

Reply:
I'm looking at this property and it's related to IME. Does it even have to be public? It could be an internal CLR property, or a `DirectProperty` with an internal CLR setter.

Reply:
TextPresenter doesn't know anything about the IME implementation so it should be treated as any other property. In reality TextPresenter itself is just some implementation detail.